src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -973,7 +1112,24 @@ def _traverse_nodes(self, inputs, context: FlowExecutionContext) -> Tuple[dict,
         if self._should_use_async():
             flow_logger.info("Start executing nodes in async mode.")
             scheduler = AsyncNodesScheduler(self._tools_manager, self._node_concurrency)
-            nodes_outputs, bypassed_nodes = asyncio.run(scheduler.execute(batch_nodes, inputs, context))
+            # Use async_run_allow_running_loop to avoid calling asyncio.run from a loop when running a notebook.
+            nodes_outputs, bypassed_nodes = async_run_allowing_running_loop(
```

Comment:
I don't think we have such scenario, why do we modify this?

Reply:
There are test cases that run this code in a jupyter notebook. If we don't use async_run_allowing_running_loop, those cases will fail.